### PR TITLE
MB-12620 remove duplicate shipment related shapes in FE code

### PR DESCRIPTION
**Pull request by felipe-lee**
## Summary

This PR merges our MTOShipment and PPMShipment shapes on the FE so we don't have duplicate shapes. I posted about [merging our two PPMShipment shapes that we have on the front-end](https://ustcdp3.slack.com/archives/C0250Q8469K/p1653438676018659), but didn't get to it in a [past PR](https://github.com/transcom/mymove/pull/8684) to avoid making that one too big. 

<details>
<summary>Change Log</summary>

### Main Changes

- [x] Merge PPMShipment shapes
- [x] Merge MTOShipment shapes

### Other Changes

- [x] Add more PPMShipment status types to the `ppmShipmentStatuses` constant
- [x] Merge MTOAgent shapes
- [x] Merge DutyLocation shapes
- [x] Sort local imports for files i updated (hopefully this doesn't break any CSS...)
- [x] Remove unused customer shapes
- [x] Fix a few invalid test data attributes that were flagged with the shape merges/updates.

</details>

### Checking for Console Errors

<details>
<summary>💻 Run Jest Tests</summary>

#### Terminal

Check that there are no console errors/warnings related to this change. There are a bunch for missing icons, but that's separate. There's also one that says `malformed api error response`, but that's also pre-existing.

```sh
make client_test
```

</details>

### Checking Storybook

I updated a story file to remove some attributes, but that triggered the need for design review. The stories in question can be viewed here (note that the attributes I changed weren't even being used so there should be no visible changes):

[Circle CI Storybook link](https://output.circle-artifacts.com/output/job/2da47ea0-4b95-456b-a2d5-de5d5ad31b7c/artifacts/0/storybook/index.html)

## Verification Steps for Author

These are to be checked by the author.

- [x] Request review from a member of a different team.
- [x] Have the Jira acceptance criteria been met for this change?
    - This one isn't a direct ticket fix, it's more of a related fix.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] There are no new console errors in the test output

### CODE REVIEW

File: src/types/shipment.js [hunk 1]
```diff
@@ -15,64 +16,99 @@ export const ShipmentOptionsOneOf = PropTypes.oneOf([
...
+  deletedAt: string,
+  sitLocation: LOCATION_TYPES_ONE_OF,
+  sitEstimatedWeight: number,
+  sitEstimatedEntryDate: string,
+  sitEstimatedDepartureDate: string,
+  sitEstimatedCost: number,
+  eTag: string,
+});
+
+export const ShipmentShape = shape({
```
Comment (duncan-truss) [inline src/types/shipment.js:75]: This looks like it's missing an eTag property

File: src/types/shipment.js [hunk 2]
```diff
@@ -15,64 +16,99 @@ export const ShipmentOptionsOneOf = PropTypes.oneOf([
...
+  ppmShipmentStatuses.NEEDS_ADVANCE_APPROVAL,
+  ppmShipmentStatuses.NEEDS_PAYMENT_APPROVAL,
+  ppmShipmentStatuses.PAYMENT_APPROVED,
+]);
+
+export const PPMShipmentShape = shape({
+  id: string,
+  shipmentId: string,
+  createdAt: string,
+  updatedAt: string,
```
Comment (duncan-truss) [inline src/types/shipment.js:42]: I'm not sure we should be exposing updatedAt and deletedAt fields.  UpdatedAt is covered by the eTag unless we are doing some sort of sorting, but I think I've only seen us sort by creation date.  They aren't hurting anything but just a suggestion.

File: src/types/shipment.js [hunk 3]
```diff
@@ -15,64 +16,99 @@ export const ShipmentOptionsOneOf = PropTypes.oneOf([
   SHIPMENT_OPTIONS.PPM,
 ]);
 
-export const ShipmentShape = PropTypes.shape({
+export const ShipmentStatusesOneOf = oneOf([
+  shipmentStatuses.DRAFT,
```
Comment (duncan-truss) [inline src/types/shipment.js:20]: If you did `Object.values(shipmentStatuses)` then we wouldn't need to worry about edits to the other list not being reflected here.

File: src/types/shipment.js (hunk 1 above)
Comment (felipe-lee) [reply src/types/shipment.js:75]: Ah, good catch, I'll add that.

File: src/types/shipment.js (hunk 2 above)
Comment (felipe-lee) [reply src/types/shipment.js:42]: I mainly made this match the payload that we have set in the yaml, but I can double check for usage and if we don't use them, I'll remove them from the yaml and here.

File: src/types/shipment.js (hunk 3 above)
Comment (felipe-lee) [reply src/types/shipment.js:20]: Oh, good idea, I'll do that.

**Review by duncan-truss — APPROVED** [review]
I worked through the apps and didn't see any propType errors that seemed related to your changes.

One I did see on the move task order page of the TOO was around the [sitStatus property](https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/swagger-def/definitions/MTOShipment.yaml#L114) of the shipment.  I noticed that was never added to the shipment shape but is used here https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx#L149.

Then for the TIO there is an [ad-hoc shipment shape defined](https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/src/components/Office/WeightSummary/WeightSummary.jsx#L68) which defines some props as required like calculatedBillableWeight, which isn't in the shape either.

**Comment by felipe-lee** [comment]
> I worked through the apps and didn't see any propType errors that seemed related to your changes.

🎉 

> One I did see on the move task order page of the TOO was around the [sitStatus property](https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/swagger-def/definitions/MTOShipment.yaml#L114) of the shipment. I noticed that was never added to the shipment shape but is used here
> 
> https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx#L149
> .

Interesting, it looks like for that one the `sitStatus` is being pulled out in the parent component from the shipment and then passed in as a separate prop. I'll add sitStatus as a field to ShipmentShape, but won't do refactoring of the components in this PR since it's a little beyond the scope of what I was aiming for with this ticket. We probably have a lot of cleanup to do throughout where we do this type of thing though. I noticed several components do this where they'll pull out parts of a shipment and pass those in separately.

> Then for the TIO there is an [ad-hoc shipment shape defined](https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/src/components/Office/WeightSummary/WeightSummary.jsx#L68) which defines some props as required like calculatedBillableWeight, which isn't in the shape either.

I'll add the props to the ShipmentShape, but do wonder what we should do about things like this where we require specific fields for certain components but not others.

**Comment by duncan-truss** [comment]
> > I worked through the apps and didn't see any propType errors that seemed related to your changes.
> 
> 🎉
> 
> > One I did see on the move task order page of the TOO was around the [sitStatus property](https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/swagger-def/definitions/MTOShipment.yaml#L114) of the shipment. I noticed that was never added to the shipment shape but is used here
> > https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx#L149
> > 
> > .
> 
> Interesting, it looks like for that one the `sitStatus` is being pulled out in the parent component from the shipment and then passed in as a separate prop. I'll add sitStatus as a field to ShipmentShape, but won't do refactoring of the components in this PR since it's a little beyond the scope of what I was aiming for with this ticket. We probably have a lot of cleanup to do throughout where we do this type of thing though. I noticed several components do this where they'll pull out parts of a shipment and pass those in separately.
> 
> > Then for the TIO there is an [ad-hoc shipment shape defined](https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/src/components/Office/WeightSummary/WeightSummary.jsx#L68) which defines some props as required like calculatedBillableWeight, which isn't in the shape either.
> 
> I'll add the props to the ShipmentShape, but do wonder what we should do about things like this where we require specific fields for certain components but not others.

@felipe-lee Yea I've seen us define shaped like that for the required address shape.  We might be able to write a wrapper that takes a shape and required props array and then toggles certain properties required.  Or there could be another library out there with richer features to adopt like https://www.npmjs.com/package/airbnb-prop-types.

**Comment by felipe-lee** [comment]
> > > I worked through the apps and didn't see any propType errors that seemed related to your changes.
> > 
> > 
> > 🎉
> > > One I did see on the move task order page of the TOO was around the [sitStatus property](https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/swagger-def/definitions/MTOShipment.yaml#L114) of the shipment. I noticed that was never added to the shipment shape but is used here
> > > https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx#L149
> > > 
> > > .
> > 
> > 
> > Interesting, it looks like for that one the `sitStatus` is being pulled out in the parent component from the shipment and then passed in as a separate prop. I'll add sitStatus as a field to ShipmentShape, but won't do refactoring of the components in this PR since it's a little beyond the scope of what I was aiming for with this ticket. We probably have a lot of cleanup to do throughout where we do this type of thing though. I noticed several components do this where they'll pull out parts of a shipment and pass those in separately.
> > > Then for the TIO there is an [ad-hoc shipment shape defined](https://github.com/transcom/mymove/blob/4d84867f63c35f1da2101501f7330c55ad749aa9/src/components/Office/WeightSummary/WeightSummary.jsx#L68) which defines some props as required like calculatedBillableWeight, which isn't in the shape either.
> > 
> > 
> > I'll add the props to the ShipmentShape, but do wonder what we should do about things like this where we require specific fields for certain components but not others.
> 
> @felipe-lee Yea I've seen us define shapes like that for the required address shape. We might be able to write a wrapper that takes a shape and required props array and then toggles certain properties required. Or there could be another library out there with richer features to adopt like https://www.npmjs.com/package/airbnb-prop-types.

Oh nice, that would be a nice thing to have so we could still have shared overall definitions while being able to specify what a given component actually needs.

**Comment by felipe-lee** [comment]
😞 I wish the codeowners stuff was a bit better about knowing if they actually need to review things...I updated a few attributes on the test data in a stories file and need design review now but nothing changed visually

**Review by scarequotes — APPROVED** [review]
groovy

**Pull request merged**